app/routes/verify.tsx

```diff
@@ -0,0 +1,83 @@
+import { CheckCircleIcon, XCircleIcon } from '@heroicons/react/outline';
+import { useLoaderData, useNavigate } from '@remix-run/react';
+import { DataFunctionArgs } from '@remix-run/server-runtime';
+import { useEffect, useState } from 'react';
+import { ErrorCode } from '~/generated/graphql';
+import { verifyCustomerAccount } from '~/providers/account/account';
+
+export async function loader({ request }: DataFunctionArgs) {
+    const url = new URL(request.url);
+    const token = url.searchParams.get('token');
+    if (!token) {
+        return {
+            errorCode: ErrorCode.VerificationTokenInvalidError,
+            message: 'Verification token was not provided!',
+        };
+    }
+
+    const result = await verifyCustomerAccount({ request }, token);
```

Comment:
The `verifyCustomerAccount` will actually automatically create a new authenticated session for the customer upon success, i.e. it will log them in. This is currently not being handled in this implementation, so the customer then has to manually log in after verification. You'll need to do something similar to the sign-in.tsx action.

Reply:
The solution I suggest works, but at least according to me it's kind of hacky. The loader function verifies the token and returns the result. In case of success the result contains the headers as a JSON string. After 5 seconds, a hidden submit button is clicked, which triggers an action. The action receives the headers JSON from a hidden input and passes it to the redirect function. Best I could think of!

Reply:
In response to: https://github.com/vendure-ecommerce/storefront-remix-starter/pull/17#issuecomment-1295117515

This is how we implemented it in our loader function
```typescript
import { redirect } from '@remix-run/server-runtime';

export async function loader({ request }: DataFunctionArgs) {
        ...
        const success =
            result.verifyCustomerAccount.__typename == 'CurrentUser';
        if (success) {
            return redirect('/account', { headers: result._headers });
        }
}
```

This should work, because the `headers` variable has a `Set-Cookie` field set by the requester function in the `graphqlWrapper.ts`: https://github.com/vendure-ecommerce/storefront-remix-starter/blob/master/app/graphqlWrapper.ts#L78

Reply:
Hey, the implemented solution still makes use of the `Set-Cookie` header, returned from the mutation. The reason I implemented it like this is because it would be good UX to display a success message for several seconds, before redirecting. So what I basically did was verify the token, get the headers, hold them on the client side for 5 seconds and then pass them to the submit action, which handles the redirect.

I'm sure there's a better way to do it, and I'll look into it.
Cheers!

Reply:
My apologies, I was just glossing over things. I personally think just redirecting the user to the account dashboard is sufficient as feedback, as it doubles as a sign-in. If you intend to leave a timeout based redirect in, you can still just return the session cookie in the loader function:

```typescript
return json({
            /* Object which carries something like a success signal */
        },
            {
                headers: result._headers
            });
    }
```
see https://remix.run/docs/en/v1/api/conventions#returning-response-instances

This would eliminate the need to store and later send the session cookie again, and you just need your time-based redirect logic.